qdarkstyle/utils.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""QDarkStyle is a dark stylesheet for Python and Qt applications."""
+
+# Standard library imports
+import os
+
+# Third party imports
+import qtsass
+
+# Local imports
+from qdarkstyle.qss import PATH_SCSS_MAIN, PATH_SCSS_VARIABLES, VARIABLES
+
+# Constants
+_HEADER_SCSS = '''// ---------------------------------------------------------------------------
+//
+//    File created programmatically
+//
+//    WARNING! All changes made in this file will be lost!
+//
+//----------------------------------------------------------------------------
+'''
+_HEADER_QSS = '''/* ---------------------------------------------------------------------------
+
+    Created by the qtsass compiler
+
+    WARNING! All changes made in this file will be lost!
+
+--------------------------------------------------------------------------- */
+'''
+_HERE = os.path.abspath(os.path.dirname(__file__))
+PATH_CSS_STYLES = os.path.join(_HERE, 'style.qss')
+
+
+def _dict_to_scss(data):
+    """Create a scss variables string from a dict."""
+    lines = []
+    template = "${}: {};"
+    for key, value in sorted(data.items()):
+        lines.append(template.format(key, value))
+
+    return '\n'.join(lines)
+
+
+def _scss_to_dict(string):
+    """Parse variables and return a dict."""
+    data = {}
+    lines = string.split('\n')
+
+    for line in lines:
+        line = line.strip()
+
+        if line and line.startswith('$'):
+            key, value = line.split(':')
+            key = key[1:].strip()
+            key = key.replace('-', '_')
+            value = value.split(';')[0].strip()
+
+            data[key] = value
+
+    return data
+
+
+def _create_scss_variables():
+    """Create a scss variables file."""
+    scss = _dict_to_scss(VARIABLES)
+    data = _HEADER_SCSS + scss + '\n'
+
+    with open(PATH_SCSS_VARIABLES, 'w') as f:
+        f.write(data)
+
+
+def _create_qss():
+    """Create a styles.qss file from qtsass."""
+    qtsass.compile_filename(PATH_SCSS_MAIN, PATH_CSS_STYLES, output_style='expanded')
+
+    with open(PATH_CSS_STYLES, 'r') as f:
+        data = f.read()
+
+    data = _HEADER_QSS + data
+
+    with open(PATH_CSS_STYLES, 'w') as f:
+        f.write(data)
+
+
+def create_qss():
+    """Create variables files and run qtsass compilation."""
+    _create_scss_variables()
+    _create_qss()
```

Comment:
@dpizetta this method runs things. This is what I “intended to run on import

Thoughts?

Reply:
Every time you import you will call this function to update the qss file. When the qss file would be generated to then be compiled into .py files. (I need to familiarize with scss dynamic)

Reply:
how expensive is the qss to .py conversion?

Reply:
> how expensive is the qss to .py conversion? and is this something we can rely will work on any user machine with any of the bindings?

It is not so expansive, but it is not so easy to use the tool to convert qss > py. We were providing the compiled files to simplify the use of qdarkstyle. Each binding could generate to itself, but most of then must be installed with dev-tools to integrate the compilers. 

> I mean we don't have to do this right now if we want to keep QDarkStyle more "static" but moving forward we could use this to provide users a simple way to style apps beyond DarkStyles by changing the variables (and maybe extra ones we can define).

Agreed totally. My idea is still providing the compiled files to the default qdarkstyle values. But generating the qss using your solution with scss. Then we can work on the changeable qdarkstyle colors on the "develper/user" machines. So I think it is a good idea, for now, stay with qss and compiled files in the repo. I was reading about loading ui, qss and other files directly in the python bindings, but I'm not sure about how trustful they are. I had many problems in the past with this, including open .png files (not converted to binaries). For now we can stay with nsert .scss (repo) to generate .qss (repo) and provide compiled forms .py (repo).

Opinions?

Reply:
I will add the compilation step to the scripts (so they are compiled before doing the other conversion) and a small test and then wait for the package (qtsass) to be released so the test can work, and it can be added as a dependency of QDarkStyle. Will do that later today ;-).

After that I want to automate the SVG - PNG generation from the palette colors as well.

Reply:
The scripts in the script folder do all needed until now, just run them inside the script folder, just add the qtass to them. (I don't know if you say exactly the same thing auhau). I have all the tools installed here for qt4/qt5, I can easily generate them. 

Go ahead! Thanks a lot!